### PR TITLE
MGMT-4155: Fixing integration with AMS when auth is disabled.

### PR DESCRIPTION
**Pull request by ybettan**

### CODE REVIEW

File: internal/bminventory/inventory_test.go [hunk 1]
```diff
@@ -5629,7 +5629,7 @@ func (e eventMsgMatcher) String() string {
 	return "input contains substrings"
 }
 
-var _ = Describe("AMS subscriptions", func() {
+var _ = FDescribe("AMS subscriptions", func() {
```
Comment (osherdp) [inline internal/bminventory/inventory_test.go:5632]: Don't commit FDescribe and FIt usage

**Review by osherdp — CHANGES_REQUESTED** [review]

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -730,10 +730,13 @@ func (b *bareMetalInventory) integrateWithAMS(ctx context.Context, cluster *comm
 	}
 	log.Infof("AMS subscription %s was created for cluster %s", sub.ID(), *cluster.ID)
 	if err := b.clusterApi.UpdateAmsSubscriptionID(ctx, *cluster.ID, strfmt.UUID(sub.ID())); err != nil {
-		log.WithError(err).Errorf("Failed to update ams_subscription_id in cluster %v, rolling back cluster registration", *cluster.ID)
+		log.WithError(err).Errorf("Failed to update ams_subscription_id in cluster %v, rolling back cluster registration and AMS subscription", *cluster.ID)
 		if deregisterErr := b.clusterApi.DeregisterCluster(ctx, cluster); deregisterErr != nil {
 			log.WithError(deregisterErr).Errorf("Failed to rollback cluster %s registration", *cluster.ID)
 		}
+		if deleteSubErr := b.ocmClient.AccountsMgmt.DeleteSubscription(ctx, strfmt.UUID(sub.ID())); deleteSubErr != nil {
```
Comment (gamli75) [inline internal/bminventory/inventory.go:737]: I think that this should be done before line 734 - DeregisterCluster

File: internal/bminventory/inventory_test.go (hunk 1 above)
Comment (ybettan) [reply internal/bminventory/inventory_test.go:5632]: Thanks, it is making the CI fail anyway so I don't put too much attention to it :)

Comment (ybettan) [reply internal/bminventory/inventory_test.go]: Fixed.

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (ybettan) [reply internal/bminventory/inventory.go:734]: Changed.

**Pull request merged**